src/pages/cartpage/components/CartProductCard.js

```diff
@@ -31,7 +33,11 @@ function CartProductCard({product}){
 
                             </div>
                             <button className="button primary-blue" onClick={()=>cartDispatch({type:"REMOVE_FROM_CART",payload:product})}>Remove from cart</button>
-                            <button className="button primary-blue">Move to wishlist</button>
+                            { 
+                            itemsInWishlist.some(item => item._id === _id) ? 
+                            <Link to="/wishlist"><button className="button primary-yellow">Go to wishlist</button></Link> :
+                            <button className="button primary-blue" onClick={()=>wishlistDispatch({type:"MOVE_TO_WISHLIST",payload:product})}>Move to wishlist</button>
```

Comment:
i would suggest not to clutter your jsx with logic you can extract some part from it and implement it accordingly.